### ISSUE TITLE
.env.example

### ISSUE THREAD

**Issue by motdotla** [neftielabs/neftie#26]
Hi Neftie team,

There is a hidden feature in Dotenv Vault for creating a managed .env.example file - that is separately public.

Would this be interesting to you and your users?

<img width="1010" alt="Screen Shot 2022-05-25 at 10 05 01 PM" src="https://user-images.githubusercontent.com/3848/170419939-4a88450d-3e01-4816-b64f-362ce98fef5c.png">

**Comment by ajmnz** [comment]
> Hi Neftie team,
> 
> 
> 
> There is a hidden feature in Dotenv Vault for creating a managed .env.example file - that is separately public.
> 
> 
> 
> Would this be interesting to you and your users?
> 

Hi Mot, I was really close to emailing you the other day regarding the creation of .example files or "public" variables that anybody could pull, but I can't remember why I didn't.

That would definitely be a useful feature — not only for us, but for many OSS projects.

**Comment by motdotla** [comment]
Nice. Give it a try. The setting should be under your project's settings at https://ui.dotenv.org

It's still raw. We're using it here:

https://github.com/dotenv-org/integration-example-docker

1. Click the badge at the top of the README
2. You'll land on a page with a .env.example
3. User can pull that with npx dotenv-vault pull example - no auth necessary

**Comment by motdotla** [comment]
Unlike your other environments management of the .env.example is completely done by hand. It won't populate a new ENV just because you add it to development. That's current decision. We're open minded to changing that.

**Comment by ajmnz** [comment]
Looks good, love the env landing page too! Might as well add that badge somewhere in the readme 😉 

My two cents — I personally think `.env.example` files are _almost_ always 1:1 to the real `.env`, but with the variables stripped.
Maybe it'd make it easier to use if the Example env appeared in the Copy feature? Or a checkbox list in the Example tab, same as with the Copy with which you can enable/disable variables to be shown?

**Comment by motdotla** [comment]
> I personally think .env.example files are almost always 1:1 to the real .env

This is a good point that I agree with.

Copy feature feels wrong. Copy is an important task where the user must be careful. Adding .env.example in there makes it feel less important/dangerous. It also buries the setting.

Possibly a 'clone' from development button on the .env.example page. Or a checkbox such as 'Keep .env.example keys in sync with development keys'.. as you put it:

> Or a checkbox list in the Example tab, same as with the Copy with which you can enable/disable variables to be shown?

**Comment by ajmnz** [comment]
> Possibly a 'clone' from development

Yep. Probably this is the best thing to do, since I've also seen quite some .env.example files with some variables filled in, such as `PORT`. Or probably leave the ability to fill vars for the paid plan, since that is somewhat more advanced. 🤷🏽‍♂️

**Comment by motdotla** [comment]
More advanced, yes, but we won't do that as part of the paid plan since it helps the community. We are trying to keep paid plan features to things that immediately help a team - and keep free things available to the broader community.
